Sources/PostgresNIO/New/Connection State Machine/ConnectionStateMachine.swift

```diff
@@ -313,40 +309,107 @@ struct ConnectionStateMachine {
     
     mutating func errorReceived(_ errorMessage: PSQLBackendMessage.ErrorResponse) -> ConnectionAction {
         switch self.state {
+        case .sslRequestSent,
+             .sslNegotiated,
+             .sslHandlerAdded,
+             .waitingToStartAuthentication,
+             .authenticated,
+             .readyForQuery,
+             .error,
+             .closing:
+            return self.closeConnectionAndCleanup(.server(errorMessage))
         case .authenticating(var authState):
-            return self.avoidingStateMachineCoW { state -> ConnectionAction in
+            return self.avoidingStateMachineCoW { machine -> ConnectionAction in
                 let action = authState.errorReceived(errorMessage)
-                state = .authenticating(authState)
-                return state.modify(with: action)
+                machine.state = .authenticating(authState)
+                return machine.modify(with: action)
+            }
+        case .closeCommand(var closeStateMachine, let connectionContext):
+            guard !closeStateMachine.isComplete else {
```

Comment:
This is an affirmative completion value check, not an exceptional condition check, use `if` instead:
```suggestion
            if closeStateMachine.isComplete {
```